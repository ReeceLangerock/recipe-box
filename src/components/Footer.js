import React from "react";
import styled from "styled-components";

const FooterStyled = styled.div`
  a {
  }
`;

const I = styled.i`
  margin-left: 10px;
  color: black;
  cursor: pointer;
`;

export class Footer extends React.Component {
  handleIconClick(link) {
    window.open("https://github.com/ReeceLangerock/camper-leaderboard", "_blank");
  }
  render() {
    const currentYear = new Date().getFullYear();

    return (
      <FooterStyled className="Footer">
        Â© {currentYear}{" "}
        <a href="http://reecelangerock.com" rel="noopener noreferrer" target="_blank">
          Reece Langerock
        </a>
        <I className={`fa fa-github`} aria-hidden="false" onClick={() => this.handleIconClick()} />
      </FooterStyled>
    );
  }
}
export default Footer;
