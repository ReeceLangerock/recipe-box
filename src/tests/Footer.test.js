import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, find, render, simulate } from "enzyme";
import Footer from "./../components/Footer";
import enzymeSerializer from "enzyme-to-json/serializer";
expect.addSnapshotSerializer(enzymeSerializer);

describe("Footer", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer />, div);
  });
});

describe("Snapshot Footer", () => {
  it("renders correctly", () => {
    const component = mount(<Footer />);
    expect(component).toMatchSnapshot();
  });
});
