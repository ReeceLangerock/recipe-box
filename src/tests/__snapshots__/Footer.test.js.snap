// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Footer renders correctly 1`] = `
<Footer>
  <styled.div
    className="Footer"
  >
    <div
      className="Footer sc-bdVaJa dltBsH"
    >
      Â© 
      2017
       
      <a
        href="http://reecelangerock.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Reece Langerock
      </a>
      <styled.i
        aria-hidden="false"
        className="fa fa-github"
        onClick={[Function]}
      >
        <i
          aria-hidden="false"
          className="fa fa-github sc-bwzfXH bMynUg"
          onClick={[Function]}
        />
      </styled.i>
    </div>
  </styled.div>
</Footer>
`;
